pkg/partmgr/partmgr.go

```diff
@@ -170,15 +171,32 @@ func (p *PartitionManager) createAndUpdatePartition(t int64) (*DBPartition, erro
 	return partition, err
 }
 
-func (p *PartitionManager) updateSchema() error {
-	data, err := json.Marshal(p.cfg)
+func (p *PartitionManager) updateSchema() (err error) {
+	metricReporter, err := performance.DefaultReporterInstance()
 	if err != nil {
-		return errors.Wrap(err, "Failed to update new partition in schema file")
+		err = errors.Wrap(err, "unable to initialize performance metrics reporter")
```

Comment:
I made this change to allow Timer to get result from the inner function, since timer signature does not allow return values I have to set the error within the inner function.